engine/consensus/sealing/core.go

```diff
@@ -94,6 +86,7 @@ func NewCore(
 	sealsMempool mempool.IncorporatedResultSeals,
 	approvalConduit network.Conduit,
 	config Config,
+	getRequiredApprovalsForSealConstruction module.RequiredApprovalsForSealConstructionInstanceGetter,
```

Comment:
There are a few reasons that I'd like to inject this `module.RequiredApprovalsForSealConstructionInstanceGetter` module into the sealing core.

1. the sealing core is reading the dynamic config, and I don't want to expose the setter interface, that's why I provided only the Getter interface, and [let NodeBuilder create the setter](https://github.com/onflow/flow-go/pull/2518/files#diff-7d960671cc56ce82783f5366bad31f7108cf00270c1bb39b0e00fa57ca96334dR199) and only [let the admin use the Setter interface.](https://github.com/onflow/flow-go/pull/2518/files#diff-cb7163a6cdf8bc3d94772f989f6fb80fe289118a752166146084d8b412dc4e40R23). 
2. The admin needs [access to a global singleton object](https://github.com/onflow/flow-go/pull/2518/files#diff-cb7163a6cdf8bc3d94772f989f6fb80fe289118a752166146084d8b412dc4e40R23). However, I don't want other modules like this sealing core to depend on a global single object, which would make testing much harder. Modules like this sealing core who are using the config, should just call the Getter's method, without knowing it's a stateful object or not. 
3. Why not a `ConfigReader.GetValue(KeyForRequiredApprovals)` ? because such function would return an interface in order to support different types of dynamic configs to be returned, and I would have to cast it into `uint`, which gives up type safety check.
4. I'd like to use different locks for different dynamic configs, so that accessing different dynamic configs will have no race condition. If I use a centralized config module to manage different dynamic configs, then there must be some shared map object to acquire a lock from first, then getting a value with different lock, which is introducing race condition overhead. 

I was also thinking about more generic way, but I didn't find one that satisfies the above items, so I ended up avoid over engineering, just started simple and we can refactor later if things are more clear. 

@jordanschalm @durkmurder